Unable to resolve the jar path at first startup time

When the jar path contains a space, it throws below error:
```
[16:35:30] [Server thread/ERROR]: Error occurred while enabling Karma v1.7.7a (Is it up to date?)
java.lang.IllegalArgumentException: Illegal character in opaque part at index 28: jar:file:/D:/MCServer/1.17.1 Paper/plugins/karma-1.7.7a.jar
	at java.net.URI.create(URI.java:906) ~[?:?]
	at com.rosstail.karma.datas.FileResourcesUtils.getPathsFromResourceJAR(FileResourcesUtils.java:98) ~[karma-1.7.7a.jar:?]
	at com.rosstail.karma.datas.FileResourcesUtils.main(FileResourcesUtils.java:35) ~[karma-1.7.7a.jar:?]
	at com.rosstail.karma.Karma.initDefaultConfigs(Karma.java:174) ~[karma-1.7.7a.jar:?]
	at com.rosstail.karma.Karma.onEnable(Karma.java:60) ~[karma-1.7.7a.jar:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264) ~[patched_1.17.1.jar:git-Paper-341]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370) ~[patched_1.17.1.jar:git-Paper-341]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500) ~[patched_1.17.1.jar:git-Paper-341]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:561) ~[patched_1.17.1.jar:git-Paper-341]
	at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:475) ~[patched_1.17.1.jar:git-Paper-341]
	at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:730) ~[patched_1.17.1.jar:git-Paper-341]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:317) ~[patched_1.17.1.jar:git-Paper-341]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1217) ~[patched_1.17.1.jar:git-Paper-341]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[patched_1.17.1.jar:git-Paper-341]
	at java.lang.Thread.run(Thread.java:831) ~[?:?]
Caused by: java.net.URISyntaxException: Illegal character in opaque part at index 28: jar:file:/D:/MCServer/1.17.1 Paper/plugins/karma-1.7.7a.jar
	at java.net.URI$Parser.fail(URI.java:2966) ~[?:?]
	at java.net.URI$Parser.checkChars(URI.java:3137) ~[?:?]
	at java.net.URI$Parser.parse(URI.java:3173) ~[?:?]
	at java.net.URI.<init>(URI.java:623) ~[?:?]
	at java.net.URI.create(URI.java:904) ~[?:?]
	... 14 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>com.github.Rosstail</groupId>
     <artifactId>karma</artifactId>
-    <version>1.7.7a</version>
+    <version>1.7.8</version>
     <packaging>jar</packaging>
 
     <distributionManagement>

--- a/src/main/java/com/rosstail/karma/datas/FileResourcesUtils.java
+++ b/src/main/java/com/rosstail/karma/datas/FileResourcesUtils.java
@@ -95,7 +95,7 @@ public class FileResourcesUtils {
         //System.out.println("JAR Path :" + jarPath);
 
         // file walks JAR
-        URI uri = URI.create("jar:file:" + jarPath);
+        URI uri = new URI("jar", "file:" + jarPath, null);
         try (FileSystem fs = FileSystems.newFileSystem(uri, Collections.emptyMap())) {
             result = Files.walk(fs.getPath(folder))
                     .filter(Files::isRegularFile)

--- a/src/main/resources/plugin.yml
+++ b/src/main/resources/plugin.yml
@@ -1,5 +1,5 @@
 name: Karma
-version: 1.7.7a
+version: 1.7.8
 description: Make your players actions have consequences !
 api-version: 1.15
 author: Rosstail

